src/components/Popover/src/PopoverLayer.vue

```diff
@@ -0,0 +1,141 @@
+<template>
+	<div>
+		<pseudo-window @blur.passive="handleBlur">
+			<pseudo-window
+				@mousedown="trackClickSrc"
+				@touchstart="trackClickSrc"
+				@click.capture="handleClick"
+				@touchend="handleClick"
+			>
+				<div
+					:id="popoverApi.target"
+					ref="portal"
+					:class="$s.PopoverLayer"
+				/>
+			</pseudo-window>
+		</pseudo-window>
+	</div>
+</template>
+
+<script>
+import { fadeInFn, fadeOutFn } from '@square/maker/utils/transitions';
+import PseudoWindow from 'vue-pseudo-window';
+import Vue from 'vue';
+import { v4 as uuid } from 'uuid';
+import { PopoverAPIKey } from './keys';
+
+const popoverMixin = {
+	inject: {
+		currentPopoverLayer: {
+			from: PopoverAPIKey,
+			default: undefined,
+		},
+	},
+
+	provide() {
+		/**
+		 * This is to avoid name collisions for the popover portal if
+		 * multiple exist at the same 'level'.
+		 */
+		const layerId = uuid();
```

Comment:
this is way overkill, lol. there's gonna be like a few popovers in the same layer at most, we don't need to use v4 uuids to prevent collisions, if we need a random string of characters we can just use `Math.random().toString(36).slice(2, 12)` which will work plenty well.

Reply:
Lol sounds good. I'll do that.